Lesson5.ipynb

```diff
@@ -29,6 +32,384 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "table = [[1, 2, 3],\n",
+    "         [4, 5, 6],\n",
+    "         [7, 8, 9]]\n",
+    "flatten = [num for row in table for num in row]\n",
+    "print(flatten)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Example 4: Given 2-dimentional list (like a table), transpose rows and columns using Nested List Comprehensions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transpose = [[row[num] for row in table] for num in range(len(table[0]))]\n",
+    "for display in transpose:\n",
+    "    print(display)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Sorting\n",
+    "\n",
+    "### Ascending - from small to big, from low to high, from a to z\n",
+    "\n",
+    "By default, all sorting are in ascending order.\n",
+    "\n",
+    "**sorted(list)** will return a new sorted list, leaving the original list unaffected.\n",
+    "\n",
+    "**list.sort()** will sorts the list **in-place**. In case list is muted, it will return None.\n",
+    "\n",
+    "### Descending - in reverse of Ascending\n",
+    "\n",
+    "Both sort() and sorted() accept additional parameter as reverse in boolean type.\n",
+    "\n",
+    "**sorted(list, reverse=True)**\n",
+    "\n",
+    "**list.sort(reverse=True)**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Example 1: Given a list of numbers, create a new sorted list in ascending order and in-place descending order."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "listOfNum = [56, 34, 65, 12, 88, 54, 99, 78]\n",
+    "\n",
+    "# new ascending order list named ascending\n",
+    "ascending = sorted(listOfNum)\n",
+    "\n",
+    "# in-place descending order\n",
+    "listOfNum.sort(reverse=True)\n",
+    "\n",
+    "print(\"Ascending: \", ascending)\n",
+    "print(\"Descending: \", listOfNum)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Example 2: Given a list of alphabets, create a new sorted list in descending order and in-place ascending order."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "listOfAlp = ['g', 'w', 's', 'a', 'k', 'e', 'q', 'd']\n",
+    "\n",
+    "# new descending order list named descending\n",
+    "descending = sorted(listOfAlp, reverse=True)\n",
+    "\n",
+    "# in-place ascending order\n",
+    "listOfAlp.sort()\n",
+    "\n",
+    "print(\"Ascending: \", listOfAlp)\n",
+    "print(\"Descending: \", descending)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Slicing\n",
+    "\n",
+    "Slicing used to extract part of a string or list.\n",
```

Comment:
"Slicing used to extract part of a string or list." should read "Slicing is used to extract part of a string or list."